File: app/src/main/java/it/cammino/risuscito/ui/RisuscitoApplication.java
package it.cammino.risuscito.ui;

import android.arch.persistence.room.Room;
import android.content.Context;
import android.graphics.drawable.Drawable;
import android.net.Uri;
import android.support.multidex.MultiDexApplication;
import android.widget.ImageView;

import com.mikepenz.iconics.IconicsDrawable;
import com.mikepenz.materialdrawer.util.AbstractDrawerImageLoader;
import com.mikepenz.materialdrawer.util.DrawerImageLoader;
import com.mikepenz.materialdrawer.util.DrawerUIUtils;

import it.cammino.risuscito.R;
import it.cammino.risuscito.database.RisuscitoDatabase;
import uk.co.chrisjenx.calligraphy.CalligraphyConfig;

public class RisuscitoApplication extends MultiDexApplication {

    @Override
    public void onCreate() {
        super.onCreate();

        RisuscitoDatabase.getInstance(this);

        CalligraphyConfig.initDefault(new CalligraphyConfig.Builder()
                        .setDefaultFontPath("fonts/Roboto-Regular.ttf")
                        .setFontAttrId(R.attr.fontPath)
//                        .addCustomStyle(TextField.class, R.attr.textFieldStyle)
                        .build()
        );

        //initialize and create the image loader logic
        /*DrawerImageLoader.init(new AbstractDrawerImageLoader() {
            @Override
            public void set(ImageView imageView, Uri uri, Drawable placeholder, String tag) {
                Picasso.with(imageView.getContext()).load(uri).placeholder(placeholder).into(imageView);
            }
            @Override
            public void cancel(ImageView imageView) {
                Picasso.with(imageView.getContext()).cancelRequest(imageView);
            }
        });*/

        //initialize and create the image loader logic
        DrawerImageLoader.init(new AbstractDrawerImageLoader() {
            @Override
            public void set(ImageView imageView, Uri uri, Drawable placeholder, String tag) {
                GlideApp.with(imageView.getContext()).load(uri).placeholder(placeholder).into(imageView);
            }

            @Override
            public void cancel(ImageView imageView) {
                GlideRequests glideRequests = GlideApp.with(imageView.getContext());
                glideRequests.clear(imageView);
            }

            @Override
            public Drawable placeholder(Context ctx, String tag) {
                //define different placeholders for different imageView targets
                //default tags are accessible via the DrawerImageLoader.Tags
                //custom ones can be checked via string. see the CustomUrlBasePrimaryDrawerItem LINE 111
                if (DrawerImageLoader.Tags.PROFILE.name().equals(tag)) {
                    return DrawerUIUtils.getPlaceHolder(ctx);
                } else if (DrawerImageLoader.Tags.ACCOUNT_HEADER.name().equals(tag)) {
                    ThemeableActivity mActivity = (ThemeableActivity) ctx;
                    return new IconicsDrawable(ctx).iconText(" ").backgroundColor(mActivity.getThemeUtils().primaryColor()).sizeDp(56);
                } else if ("customUrlItem".equals(tag)) {
                    return new IconicsDrawable(ctx).iconText(" ").backgroundColorRes(R.color.md_red_500).sizeDp(56);
                }

                //we use the default one for
                //DrawerImageLoader.Tags.PROFILE_DRAWER_ITEM.name()

                return super.placeholder(ctx, tag);
            }
        });
    }
}

File: app/src/main/java/it/cammino/risuscito/ui/RisuscitoApplication.kt
package it.cammino.risuscito.ui

import android.content.Context
import androidx.multidex.MultiDexApplication
import com.google.android.material.color.DynamicColors
import com.michaelflisar.composechangelog.statesaver.preferences.ChangelogStateSaverPreferences
import com.michaelflisar.composechangelog.statesaver.preferences.create
import it.cammino.risuscito.utils.LocaleManager
import it.cammino.risuscito.utils.extension.dynamicColorOptions
import it.cammino.risuscito.utils.extension.setDefaultNightMode


class RisuscitoApplication : MultiDexApplication() {

    val changelogStateSaver: ChangelogStateSaverPreferences by lazy {
        ChangelogStateSaverPreferences.create(this)
    }
    override fun onCreate() {
        super.onCreate()

        applicationContext.setDefaultNightMode()

        DynamicColors.applyToActivitiesIfAvailable(this, dynamicColorOptions)

    }

    override fun attachBaseContext(base: Context) {
        localeManager = LocaleManager(base)
        super.attachBaseContext(base)
    }

    companion object {
        internal val TAG = RisuscitoApplication::class.java.canonicalName
        lateinit var localeManager: LocaleManager
    }
}
